sftp/sftp.go

```diff
@@ -455,9 +498,37 @@ func (o *Object) Remote() string {
 	return o.remote
 }
 
-// Hash returns "" since SFTP (in Go or OpenSSH) doesn't support remote calculation of hashes
 func (o *Object) Hash(r fs.HashType) (string, error) {
-	return "", fs.ErrHashUnsupported
+	session, err := o.fs.sshClient.NewSession()
```

Comment:
The other remotes save the hash in the Object. rclone doesn't modify objects it creates new ones so the cache doesn't need to be invalidated.  The object could change remotely you are right - but this is the same risk as the other backends.